pkg/types/balance_type.go

```diff
@@ -0,0 +1,81 @@
+package types
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/pkg/errors"
+)
+
+type BalanceType string
+
+const (
+	BalanceTypeAvailable BalanceType = "AVAILABLE"
+	BalanceTypeLocked    BalanceType = "LOCKED"
+	BalanceTypeBorrowed  BalanceType = "BORROWED"
+	BalanceTypeInterest  BalanceType = "INTEREST"
+	BalanceTypeNet       BalanceType = "NET"
+	BalanceTypeTotal     BalanceType = "TOTAL"
+	BalanceTypeDebt      BalanceType = "DEBT"
+)
+
+var ErrInvalidBalanceType = errors.New("invalid balance type")
+
+func StrToBalanceType(s string) (b BalanceType, err error) {
+	switch strings.ToLower(s) {
```

Comment:
you can use ToUpper? and if the string matches any type vars you defined in the const, you can simply assign it

Reply:
done